src/test/kotlin/com/terraformation/backend/tracking/db/observationStore/ObservationStoreSurvivalRateCalculationTest.kt

```diff
@@ -0,0 +1,447 @@
+                    statusId = RecordedPlantStatus.Dead,
+                ),
+                RecordedPlantsRow(
+                    certaintyId = RecordedSpeciesCertainty.Other,
+                    gpsCoordinates = point(101),
+                    speciesName = "Who knows",
+                    statusId = RecordedPlantStatus.Existing,
+                ),
+            )
+    store.completePlot(observationId, plotId, emptySet(), "Notes1", observedTime, plot1Plants)
+
+    val plot1SurvivalRates: Map<Any, Map<SpeciesId, Double?>> =
+        mapOf(
+            plotId to
+                mapOf(
+                    speciesId1 to (100.0 * 9 / 15),
+                    speciesId2 to (100.0 * 18 / 23),
+                    speciesId3 to (100.0 * 27 / 31),
+                )
+        )
+    val survivalRates1 =
+        mapOf(
+            speciesId1 to (100.0 * 9 / (15 + 9)),
+            speciesId2 to (100.0 * 18 / (23 + 19)),
+            speciesId3 to (100.0 * 27 / 31),
+        )
+
+    assertSurvivalRates(
+        listOf(
+            plot1SurvivalRates,
+            mapOf(subzoneId to survivalRates1),
+            mapOf(zoneId to survivalRates1),
+            mapOf(plantingSiteId to survivalRates1),
+        ),
+        "Plot 1 survival rates",
+    )
+
+    val plot2Plants =
+        createPlantsRows(
+            mapOf(speciesId1 to 10, speciesId2 to 17, speciesId3 to 25),
+            RecordedPlantStatus.Live,
+        ) +
+            createPlantsRows(
+                mapOf(speciesId1 to 0, speciesId2 to 1, speciesId3 to 1),
+                RecordedPlantStatus.Dead,
+            ) +
+            listOf(
+                RecordedPlantsRow(
+                    certaintyId = RecordedSpeciesCertainty.Other,
+                    gpsCoordinates = point(102),
+                    speciesName = "Who knows",
+                    statusId = RecordedPlantStatus.Existing,
+                )
+            )
+    store.completePlot(observationId, plot2, emptySet(), "Notes2", observedTime, plot2Plants)
+
+    val plot2SurvivalRates: Map<Any, Map<SpeciesId, Double?>> =
+        mapOf(
+            plot2 to
+                mapOf(
+                    speciesId1 to (100.0 * 10 / 9),
+                    speciesId2 to (100.0 * 17 / 19),
+                    speciesId3 to null,
+                )
+        )
+    val survivalRates1And2 =
+        mapOf(
+            speciesId1 to (100.0 * (9 + 10) / (15 + 9)),
+            speciesId2 to (100.0 * (17 + 18) / (23 + 19)),
+            speciesId3 to (100.0 * (27 + 25) / 31),
+        )
+    assertSurvivalRates(
+        listOf(
+            plot1SurvivalRates + plot2SurvivalRates,
+            mapOf(subzoneId to survivalRates1And2),
+            mapOf(zoneId to survivalRates1And2),
+            mapOf(plantingSiteId to survivalRates1And2),
+        ),
+        "Plots 1 and 2 survival rates",
+    )
+  }
+
+  @Test
+  fun `survival rate is calculated for a site using all data`() {
```

Comment:
This test was created using data from the spreadsheet, hence why there's so much extra data

Reply:
Would it make sense to use the actual spreadsheet (or a CSV export of it) here? We do that in some of the other rate-calculation tests. The main reason I wrote the tests that way was so that if the sheet gets changed because we decided to change the calculations, we can just re-export it and the test code won't have to change.

But obviously that requires writing code to process the CSVs, which may be more trouble than it's worth in this case.

Reply:
It's definitely more work, but with how complex this stuff is I rather like that idea. I was super nervous already while copying all the data of accidentally touching one of the cells because even `undo` doesn't go back to previous random values so every value would have changed. I saved several screenshots during it ha. I'll look into this

Reply:
The more I think about this, the more I lean towards waiting for now. This scenario already covers quite a bit of different situations, so I'll wait till the second time around to write the parser stuff for it. Might be worth extracting from ObservationResultsStoreTest for reuse.